kafka/producer/base.py

```diff
@@ -27,6 +27,9 @@
 
 STOP_ASYNC_PRODUCER = -1
 
+# unlimited
+ASYNC_QUEUE_MAXSIZE = 0
```

Comment:
agree with @bcwalrus 

i tend to fall into "prevent ability to do stupid things", so i'd prefer to put a limit by default, so that we prevent stupid/accidental behaviour in the future. as @bcwalrus says, if they truly want _unlimited_, they should pass in 0 in that case